frontend/src/app/providers/router/ui/RequireAuth.tsx

```diff
@@ -0,0 +1,19 @@
+import type { JSX } from "react";
+import { Navigate, useLocation } from "react-router";
+import { RoutePaths } from "../../../../shared/config/routeConfig/routeConfig";
+
+export function RequireAuth({ children }: { children: JSX.Element }) {
```

Comment:
```suggestion
export function RequireAuth({ children }: PropsWithChildren) {
```
определяемся, что будем использовать 

```suggestion
const RequireAuth: FC<PropsWithChildren> = ({ children } => {
```
и везде пользуем одинаково